gapis/replay/executor.go

```diff
@@ -112,6 +114,11 @@ func (e executor) HandlePostData(ctx context.Context, postData *gapir.PostData)
 // HandleNotification implements gapir.ReplayResponseHandler interface.
 func (e executor) HandleNotification(ctx context.Context, notification *gapir.Notification) error {
 	e.handleNotification(notification)
+	// However notifications are handled, never survive a fatal error
+	errMsg := notification.GetErrorMsg()
+	if errMsg != nil && errMsg.Severity == severity.Severity_FatalLevel {
+		return log.Errf(ctx, fmt.Errorf("%s", errMsg.Msg), "Received a fatal error during replay")
```

Comment:
`errors.New(errorMsg.Msg)` if you want to wrap the string into an error. However, that doesn't make much sense here, so why not just `return fmt.Errorf("Replay failed: %v", errMsg.Msg)`?

Reply:
Done. I felt like we had to use log.Errf() to pass the `ctx` context here.